### DISCUSSION TITLE
Does the compat package supports hash routeing

### DISCUSSION THREAD

**Discussion by kshanmugharaj** [remix-run/react-router#8755] [v5 to v6 Migration]
We have the setup similar to this

```
<Router history={history}>
    <CompatRouter>
        <Route path="/" render={renderRoutes} />
    </CompatRouter>
</Router>
```

now with the addition of `CompatRouter` the redirection is not happening. But when I modify `<Router history={history}>` to `<BrowserRouter />` its working. So hash based routing is not supported or am I doing some mistake?

**Comment by kshanmugharaj** [comment]
So using the `<HashRouter />` is working as expected. Does `<Router history={history}>` and just using `<HashRouter />` are same? I'm not sure why it was written that way since it's a old code base.

**Comment by ryanflorence** [comment]
`<CompatRouter>` is accessing your history with v5 `useHistory`, so I would expect it to work with `<Router history>` just the same as `<HashRouter>` or `<BrowserRouter>`.
